File: math/v1/chem/pressure/osmotic_pressure.py
#!/usr/bin/env python3
# chem/osmotic_pressure.py

"""Osmotic Pressure:

Π = osmotic pressure

Π = MRT
Π = M*r*t

M = molarity (M): (mol / volume)
t = Kelvins temp (T)
r = R constant

mass, weight, volume, temp (in C)

# osmotic_pressure(m, w, v, t, pressure_unit) -> float

"""

from .r import r, print_R
from ..solutions import molarity
from ..units import mol, kelvins

def osmotic_pressure(m, w, v, t, pressure_unit):
    return (
        kelvins(t) # T
        * r(pressure_unit) # R
        * molarity( # M:
            mol(m, w), # mols
            v          # liters
        )
    )

if __name__ == "__main__":
    from util import get_vars, alias

    print("Π = MRT")
    print("Π = ((mass/mw)/v)RT")
    print_R()
    print("")

    vars = get_vars({
        "mass":float,
        "mw":float,
        "volume":float,
        "temp (C)":float,
        "pressure type":str
    })

    print(osmotic_pressure(
        **alias(vars, {
            "mass":"m",
            "mw":"w",
            "volume": "v",
            "temp (C)": "t",
            "pressure type": "pressure_unit"
        })
    ))

File: chem/util/quick-calc/osmoticPressure.js
const getVars = require("../getvars")
const parse = require("../parsers")

const { R } = require("../../vars/constants")
const { molarity } = require("../../formulas/solutions")
const osmoticPressure = require("../../vars/osmotic-pressure")
const mol = require("../../vars/mol")

module.exports = () => {
  args = getVars({
    mass: parse.float,
    mw: parse.float,
    volume: parse.float,
    temperature: parse.float,
    ptype: parse.string
  })

  let r = R.get(args.ptype)
  let M = molarity(mol(args.mass, args.mw), args.volume)

  let ans = osmoticPressure(M, r, args.temperature)
  console.log(`π = ${ans} ${args.ptype}`)
}
